### ISSUE TITLE
Cannot parse disabled pub type

### ISSUE THREAD

**Issue by P-E-P** [Rust-GCC/gccrs#2666]
Cannot parse disabled pub type in trait.

I tried this code:

```rust
fn main() {}

#[cfg(FALSE)]
trait T {
    pub type X;
}
```

I expected to see this happen: Compiles correctly.

Instead, this happened: Does not compile.

```
./test.rs:5:5: error: expecting ‘!’ but ‘pub’ found
    5 |     pub type X;
      |     ^~~
./test.rs:5:5: error: failed to parse trait item in trait
./test.rs:5:5: error: failed to parse item in crate
```

### Meta

- c7728c8020334fecb47789163ba94326aab6208c

**Comment by P-E-P** [comment]
> i wanna solve this issue

Sure @officialhemant511 ! I can assign it to you.

>     * The issue with the code you provided is that trait items, including associated types like type X, cannot be marked as pub. The pub keyword specifies the item's visibility, but associated types in traits are always public by default. Therefore, you can simply remove the pub keyword from the declaration of type X:

The problem here does not lie in the code in this code but rather in the compiler. `rustc` can compile this code without any error ([link to the playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=69b8fad57aa1429b3e46328a1b1d6cd7)).

It compiles because the line has been disabled thanks to `#[cfg(FALSE)]`. The step that removes part of the AST with those `cfg` directives is called "cfg stripping" in gccrs. This step relies on the ast, this means we shall refer to rust syntax reference:

Syntax
AssociatedItem :
   [OuterAttribute](https://doc.rust-lang.org/beta/reference/attributes.html)* (
         [MacroInvocationSemi](https://doc.rust-lang.org/beta/reference/macros.html#macro-invocation)
      | ( [Visibility](https://doc.rust-lang.org/beta/reference/visibility-and-privacy.html)? ( [TypeAlias](https://doc.rust-lang.org/beta/reference/items/type-aliases.html) | [ConstantItem](https://doc.rust-lang.org/beta/reference/items/constant-items.html) | [Function](https://doc.rust-lang.org/beta/reference/items/functions.html) ) )
   )

from https://doc.rust-lang.org/beta/reference/items/associated-items.html, here `pub type` fall in the type alias category. 

Note that from the syntax pov, this visibility modifier is expected even if the language reject it at a later stage with the [error 0449](https://doc.rust-lang.org/error_codes/E0449.html).

The error message from gccrs states that it originates from the parsing stage of the compiler. This means the AST has not been finished yet. We should be able to parse this syntax, even if rejected later, so it could be eliminated by the cfg stripper.

### To sum up

The parser shall be fixed to accept a visibility modifier. The problem probably is in the member function `parse_trait_item` in [rust-parse-impl.h](https://github.com/Rust-GCC/gccrs/blob/master/gcc/rust/parse/rust-parse-impl.h#L4970). The function does not expect `pub` and falls to the default case label, which s expecting a macro.

This means we should probably modify the `TraitIemType` ast node to store the visibility and reject the line at a later stage.

Do you still want to me to assign you on the issue ? I hope this didn't break your confidence or anything.
